### PR TITLE
Refactor, get rid of global gs object, and fix naming

### PR DESCRIPTION
**Pull request by belochub**
This depends on https://github.com/metarhia/globalstorage/pull/297

### CODE REVIEW

File: gs.js [hunk 1]
```diff
@@ -86,44 +101,43 @@ class GlobalStorage extends lib.StorageProvider {
...
-    return id;
+  //   category - <string>, category to select the records from
+  //   query - <Object>, fields conditions
+  //
+  // Returns: <Cursor>
+  select(category, query) {
+    return this.local.select(category, query);
   }
 
   // Get server for id
```
Comment (lundibundi) [inline gs.js]: I think this comment is incorrect.

Comment (belochub) [reply gs.js]: Should we maybe remove this method completely?

File: gs.js [hunk 2]
```diff
@@ -14,24 +14,39 @@ const submodules = [
...
     this.remotes = {};
     this.active = false;
     this.offline = true;
     this.nextId = 0;
     this.categories = {};
-    this.serverBitmask = ids.serverBitmask;
-    this.serverSuffix = ids.serverSuffix;
-    this.systemSuffix = ids.systemSuffix;
-    this.systemBitmask = ids.systemBitmask;
+    this.schema = schema;
```
Comment (nechaido) [inline gs.js]: This leads to a recursive dependency: `Metaschema` need `gs` instance to properly load Actions whilst `GlobalStorage` needs `Metaschema` instance for initialization.
IMHO, we should initialize schema on `open`.
@tshemsedinov @lundibundi @belochub WDYT?

Comment (tshemsedinov) [reply gs.js]: Better solution: we may remove class `GlobalStorage` from `gs.js`, move all code from those class to abstract `StorageProvider` or to utils, export factory `providerFactory` from `gs.js`, so we will not pass `gs` instance to providers constructor or to `open`, we will use `this` instead. Moreover we will remove recursive abstraction what we have now. It's not obvious that `GlobalStorage` extends `StorageProvider` and wraps multiple `StorageProvider` implementations at a same time.

File: lib/provider.js [hunk 3]
```diff
@@ -137,6 +157,68 @@ class StorageProvider {
...
+  // Returns: <common.Uint64>
+  getSystemSuffix(id) {
+    return Uint64.and(id, this.systemBitmask);
+  }
+
+  // Check whether data with given id is stored on this system
+  //   id - <common.Uint64>
+  //
+  // Returns: <boolean>
+  curSystem(id) {
```
Comment (lundibundi) [inline lib/provider.js]: I'd prefer something more descriptive, i.e.
```suggestion
  isOnCurrentSystem(id) {
```
```suggestion
  isOnCurSystem(id) {
```
And the same for `curServer`.

Comment (belochub) [reply lib/provider.js]: It was moved here from `gs.js`, I didn't change the naming. It's also named in accordance with https://github.com/metarhia/globalstorage/issues/185.

Comment (lundibundi) [reply lib/provider.js]: I know that it was this way before. I'm suggesting to rename it while we are still at it (and refactoring a lot of things).

Comment (belochub) [reply lib/provider.js]: I think @tshemsedinov should decide on that since he was the one to introduce the naming.

Comment (lundibundi) [reply lib/provider.js]: I think we can land this as is and I'll open a PR with this change later.

**Comment by belochub** [comment]
Landed in https://github.com/metarhia/globalstorage/commit/30b3387dff419254f5dd09b346c44fabec7e8510.

**Pull request closed without merging**